test/transcoding/annotate_attribute.ll

```diff
@@ -110,64 +110,64 @@ entry:
-  store i32 0, i32* %1, align 4, !tbaa !9
+  %f1 = getelementptr inbounds %struct.bar, ptr %s1, i32 0, i32 0
+  %1 = call ptr @llvm.ptr.annotation.p0.p0(ptr %f1, ptr @.str.4, ptr @.str.1, i32 8, ptr undef)
+  store i32 0, ptr %1, align 4, !tbaa !9
   ; CHECK-LLVM: %[[FIELD2:.*]] = getelementptr inbounds %struct.bar, ptr %{{[a-zA-Z0-9]+}}, i32 0, i32 1
   ; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0{{.*}}%[[FIELD2]]{{.*}}[[STR5]]
-  %f2 = getelementptr inbounds %struct.bar, %struct.bar* %s1, i32 0, i32 1
-  %2 = call i8* @llvm.ptr.annotation.p0i8.p0i8(i8* %f2, i8* getelementptr inbounds ([4 x i8], [4 x i8]* @.str.5, i32 0, i32 0), i8* getelementptr inbounds ([23 x i8], [23 x i8]* @.str.1, i32 0, i32 0), i32 9, i8* undef)
-  store i8 0, i8* %2, align 4, !tbaa !13
+  %f2 = getelementptr inbounds %struct.bar, ptr %s1, i32 0, i32 1
+  %2 = call ptr @llvm.ptr.annotation.p0.p0(ptr %f2, ptr @.str.5, ptr @.str.1, i32 9, ptr undef)
+  store i8 0, ptr %2, align 4, !tbaa !13
   ; CHECK-LLVM: %[[FIELD3:.*]] = getelementptr inbounds %struct.bar, ptr %{{[a-zA-Z0-9]+}}, i32 0, i32 2
   ; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0{{.*}}%[[FIELD3]]{{.*}}[[STR6]]
-  %f3 = getelementptr inbounds %struct.bar, %struct.bar* %s1, i32 0, i32 2
-  %3 = bitcast float* %f3 to i8*
-  %4 = call i8* @llvm.ptr.annotation.p0i8.p0i8(i8* %3, i8* getelementptr inbounds ([4 x i8], [4 x i8]* @.str.6, i32 0, i32 0), i8* getelementptr inbounds ([23 x i8], [23 x i8]* @.str.1, i32 0, i32 0), i32 9, i8* undef)
-  %5 = bitcast i8* %4 to float*
-  store float 0.000000e+00, float* %5, align 4, !tbaa !14
+  %f3 = getelementptr inbounds %struct.bar, ptr %s1, i32 0, i32 2
+  %3 = bitcast ptr %f3 to ptr
+  %4 = call ptr @llvm.ptr.annotation.p0.p0(ptr %3, ptr @.str.6, ptr @.str.1, i32 9, ptr undef)
+  %5 = bitcast ptr %4 to ptr
+  store float 0.000000e+00,ptr %5, align 4, !tbaa !14
   ; CHECK-LLVM: %[[FIELD4:.*]] = getelementptr inbounds %struct.bar, ptr %{{[a-zA-Z0-9]+}}, i32 0, i32 3
   ; CHECK-LLVM: call ptr @llvm.ptr.annotation.p0{{.*}}%[[FIELD4]]{{.*}}[[STR8]]
-  %f4 = getelementptr inbounds %struct.bar, %struct.bar* %s1, i32 0, i32 3
-  %6 = call i8* @llvm.ptr.annotation.p0i8.p0i8(i8* %f4, i8* getelementptr inbounds ([22 x i8], [22 x i8]* @.str.7, i32 0, i32 0), i8* getelementptr inbounds ([23 x i8], [23 x i8]* @.str.1, i32 0, i32 0), i32 9, i8* bitcast ({ i32, i32, i32 }* @.args.1 to i8*))
-  store i8 0, i8* %6, align 4, !tbaa !13
-  %7 = bitcast %struct.bar* %s1 to i8*
-  call void @llvm.lifetime.end.p0i8(i64 12, i8* %7) #4
+  %f4 = getelementptr inbounds %struct.bar, ptr %s1, i32 0, i32 3
+  %6 = call ptr @llvm.ptr.annotation.p0.p0(ptr %f4, ptr @.str.7, ptr @.str.1, i32 9, ptr @.args.1)
+  store i8 0, ptr %6, align 4, !tbaa !13
+  %7 = bitcast ptr %s1 to ptr
+  call void @llvm.lifetime.end.p0(i64 12, ptr %7) #4
```

Comment:
Can it be simplified to ?
```suggestion
  call void @llvm.lifetime.end.p0(i64 12, ptr %s1) #4
```

Reply:
Yes; I've updated this and other occurrences.